pkg/collector/node.go

```diff
@@ -75,12 +79,31 @@ func (n *NodeCollector) Collect(ch chan<- prometheus.Metric) {
 		// Desc{fqName: "node_network_transmit_bytes_total", help: "Network device statistic transmit_bytes.", constLabels: {}, variableLabels: [device]}
 		// this is ugly but currently all we can do
 		d := strings.ToLower(m.Desc().String())
-		for _, s := range whitelist {
-			if strings.Contains(d, fmt.Sprintf(`fqname: "%s"`, s)) {
-				ch <- m
+		for _, s := range metricWhitelist {
+			if !strings.Contains(d, fmt.Sprintf(`fqname: "%s"`, s)) {
+				continue
+			}
+			if strings.Contains(d, "node_network") && !validNetwork(m) {
```

Comment:
would it be better to test just the prefix?

Reply:
It would have been a better choice, but it's all the same at this point :)